client/src/jobs/job.py

```diff
@@ -66,6 +56,23 @@ class ImageOptions(BaseModel):
         protected_namespaces=(),
     )
 
+    def __init__(self, **data: Any) -> None:
+        spec = data.get("spec")
+        if spec is not None and not Path(spec).is_absolute():
+            if job_path := inspect.stack()[1].filename:
+                job_dir = Path(os.path.abspath(job_path)).parent
+                data["spec"] = (job_dir / Path(spec)).resolve()
```

Comment:
This entire module has too much mix of `os.path` and `Path` functions used. We should stick to one of them (`Path`, IMO).

This should be the equivalent:
```suggestion
                job_dir = Path(job_path).resolve().parent
                data["spec"] = (job_dir / Path(spec)).resolve()
```

Reply:
Ahh, damn. 

That came out of my (faulty) assumption that there was an issue with how `Path` resolves the absolute path. (In the end it was our canonicalization code -- see above). 

Agree to go with `Path`